### DISCUSSION TITLE
Improve documentation Poco::Crypto::RSAKey & Poco::Crypto::ECKey deprecated

### DISCUSSION THREAD

**Discussion by ClaudeTO80** [pocoproject/poco#3827] [Q&A] [answered]
According to their documentations, both [Poco::Crypto::RSAKey](https://docs.pocoproject.org/current/Poco.Crypto.RSAKey.html) & [Poco::Crypto::ECKey](https://docs.pocoproject.org/current/Poco.Crypto.ECKey.html)  are deprecated. I wonder if it is possible describe why such classes are deprecated and which class(es) should be used in are their replacement.

**Comment by aleks-f** [comment, marked as answer]
Anything that does not use the envelope openssl api is deprecated (as it is in [openssl, since 3.0](https://www.openssl.org/docs/man3.0/man7/migration_guide.html#Deprecation-of-Low-Level-Functions)).

In this particular case, you can use `EVPPKey`:

https://github.com/pocoproject/poco/blob/d11e48f85151ec761f959a76d4bd52d9f43a97e0/Crypto/include/Poco/Crypto/EVPPKey.h#L46

**Reply by ClaudeTO80** [reply]
@aleks-f  Thank you very much for your reply. It would be very useful if such information and links were shared in documentation. Thanks for your work.

**Comment by Mike4Online** [comment]
@aleks-f,

I am trying to migrate from `Poco::Crypto::RSAKey` to `Poco::Crypto::EVPPKey`, but I am stuck:

I can create an `EVPPKey` with the constructor:


[EVPPKey](https://docs.pocoproject.org/current/Poco.Crypto.EVPPKey.html)(
    int type,
    int param
);

Which lets me set the type to `EVP_PKEY_RSA` and set the key length.  But I cannot set the exponent to small or large, as I can when constructing an RSAKey:

[RSAKey](https://docs.pocoproject.org/current/Poco.Crypto.RSAKey.html)(
    [KeyLength](https://docs.pocoproject.org/current/Poco.Crypto.RSAKey.html#862) keyLength,
    [Exponent](https://docs.pocoproject.org/current/Poco.Crypto.RSAKey.html#867) exp
);

Also, when I need to create a Cipher from this new key, I can do so using two methods:

[Cipher](https://docs.pocoproject.org/current/Poco.Crypto.Cipher.html) * createCipher(
    const [RSAKey](https://docs.pocoproject.org/current/Poco.Crypto.RSAKey.html) & key,
    [RSAPaddingMode](https://docs.pocoproject.org/current/Poco.Crypto.CipherFactory.html#279) paddingMode = RSA_PADDING_PKCS1
);

Or

[Cipher](https://docs.pocoproject.org/current/Poco.Crypto.Cipher.html) * createCipher(
    const [EVPPKey](https://docs.pocoproject.org/current/Poco.Crypto.EVPPKey.html) & key
);

The second method, which takes an EVPPKey, offers no optional parameter to set the RSA padding mode.  The first method offers this option, but it uses the deprecated RSAKey class which I am trying to get away from.

How can I use EVPPKey for RSA encryption with a specified exponent size and RSA padding mode?

Thank you.
